module.js

```diff
@@ -0,0 +1,78 @@
+var dateUtil = (function() {
+  var privateDate = null;
+
+  function initDate(){
+    privateDate = new Date();
+    return privateDate;
+  }
+
+  function setDate(date) {
+    if(typeof date === 'undefined')
+      privateDate = initDate();
+    else if(date instanceof Date)
+      privateDate =  date;
+	else
+      privateDate =  new Date(date);
+
+    return privateDate;
+  }
+
+  function getDate(options){
+    if(!privateDate)
+      initDate();
+    if(typeof options === 'undefined' || options.format === 'milliseconds')
+      return 'Today\'s date is ' + privateDate.getTime();
+    else if(options.format === 'formatted')
+      return 'Today\'s date is ' + getMonthName() + ' ' + privateDate.getDate() +','
+      + privateDate.getFullYear();
+  }
+
+  function getDayName(){
+    var days = ['Sunday','Monday','Tuesday','Wednesday','Thursday','Friday','Saturday'];
+    if(!privateDate)
+      initDate();
+
+    return days[privateDate.getDay()];
+  }
+
+  function getMonthName(){
+    var months = ['January','February','March','April','May','June','July','August',
+			'September','October','November','December'];
+    if(!privateDate)
+      initDate();
+
+    return months[privateDate.getMonth()];
+  }
+
+  function isFuture(){
+    if(!privateDate)
+      initDate();
+    var today = new Date().toDateString();
+    if(privateDate.toDateString() > today)
```

Comment:
This won't do what you expect. The toDateString() method will return a formatted string which, when comparing to another string, will determine inequality based on ascii value order starting with the first character in the string. Instead, compare the millisecond values returned by the `getTime()` method, as it will return a millisecond value which lends itself to inequality checks.